src/coreclr/nativeaot/Runtime/ObjectLayout.h

```diff
@@ -131,3 +131,11 @@ static uintptr_t const STRING_BASE_SIZE = StringConstants::BaseSize;
 
 //-------------------------------------------------------------------------------------------------
 static uintptr_t const MAX_STRING_LENGTH = 0x3FFFFFDF;
+
+//-------------------------------------------------------------------------------------------------
+class WeakReference : public Object
+{
+public:
+    size_t       m_Handle;
+    uint16_t     m_IsLongReference;
```

Comment:
`bool` is uint8_t.

Reply:
Ah, right. I am thinking of removing this bool altogether (not in this change), since the handle knows its kind. 

It does needs to be `uint8_t`.

Reply:
https://github.com/dotnet/runtime/issues/75586 to follow up with removing the field